ql/src/java/org/apache/hadoop/hive/ql/exec/DataSketchesFunctions.java

```diff
@@ -53,169 +72,256 @@
   private static final String SKETCH_TO_VARIANCES = "variances";
   private static final String SKETCH_TO_PERCENTILE = "percentile";
 
-  private final Registry system;
+  private List<SketchDescriptor> sketchClasses;
```

Comment:
Can we populate this statically to avoid triggering same logic twice when we call register*?

Reply:
I don't really want to make this field static; but making the constructor private and creating a singleton might achieve the same goal (avoid constructing twice)